core/src/services/s3/backend.rs

```diff
@@ -1335,6 +1373,14 @@ impl S3Backend {
             req = req.header(CACHE_CONTROL, cache_control)
         }
 
+        // Set storage class header
+        if let Some(v) = storage_class
+            .and_then(|v| v.parse::<HeaderValue>().ok())
+            .or_else(|| self.storage_class.clone())
+        {
+            req = req.header(HeaderName::from_static(constants::X_AMZ_STORAGE_CLASS), v);
+        }
+
```

Comment:
Based on https://docs.aws.amazon.com/AmazonS3/latest/userguide/storage-class-intro.html, we only need to set the storage class header when using the `PUT Object` and `Initiate Multipart Upload` API operations.